main-command/src/main/scala/sbt/Stamped.scala

```diff
@@ -0,0 +1,74 @@
+/*
+ * sbt
+ * Copyright 2011 - 2018, Lightbend, Inc.
+ * Copyright 2008 - 2010, Mark Harrah
+ * Licensed under Apache License 2.0 (see LICENSE)
+ */
+
+package sbt
+
+import java.io.{ File => JFile }
+import java.nio.file.Path
+
+import sbt.internal.inc.Stamper
+import sbt.io.TypedPath
+import xsbti.compile.analysis.Stamp
+
+/**
+ * A File that has a compile analysis Stamp value associated with it. In general, the stamp method
+ * should be a cached value that can be read without doing any io. This can be used to improve
+ * performance anywhere where we need to check if files have changed before doing potentially
+ * expensive work.
+ */
+trait Stamped {
```

Comment:
I get a bit nervous when we expose the inners of Zinc.
Does this class need to be public, or can we keep this internal?

Reply:
I addressed this in 
https://github.com/sbt/sbt/pull/4535/commits/77baef665140cf76ad47b4a5801c1b25bd5ac4d0 (Add FileCacheEntry).